### ISSUE TITLE
Misleading error when a private trait isn't imported

### ISSUE THREAD

**Issue by arduano** [rust-lang/rust-analyzer#16594]
Hi, just spent a bit trying to figure out a confusing error message. I believe this is a rust-analyzer bug as `cargo check` provides a more neutral/vague error message, while rust-analyzer's one was misleading.

Here is the minimal reproduction:
```rs
mod foo {
    trait Trait {
        fn run(&self);
    }

    pub struct Struct;

    impl Trait for Struct {
        fn run(&self) {
            println!("Hello");
        }
    }
}

use foo::Struct;

fn test() {
    let s = Struct;
    s.run();
}
```

And here is the error message:
```
no method named `run` found for struct `Struct` in the current scope
items from traits can only be used if the trait is implemented and in scope

(location): method `run` not found for this struct
(location): `Trait` defines an item `run`, perhaps you need to implement it
```

Most of this is ok, except for the last line, which was very confusing because Trait was in fact implemented for Struct.

When making the trait public, a more reasonable error message happens, telling me to import the trait.

**rust-analyzer version**: (eg. output of "rust-analyzer: Show RA Version" command, accessible in VSCode via <kbd>Ctrl/⌘</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>)
0.3.1839-standalone

**rustc version**: (eg. output of `rustc -V`)
1.74.1

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)
N/A

**Comment by Young-Flash** [comment]
it's not a rust-analyzer issue, with rustc 1.77.0-nightly (2a3e63551 2023-12-30), we got

![image](https://github.com/rust-lang/rust-analyzer/assets/71162630/953c8ca3-dd16-43be-b070-f0ac9a722e8f)

while with rustc 1.74.1 we got

![image](https://github.com/rust-lang/rust-analyzer/assets/71162630/7afd791e-e64a-4ebc-88c9-0829cac266c9)
